avro-serializer/src/main/java/io/confluent/kafka/formatter/AvroMessageFormatter.java

```diff
@@ -98,13 +107,29 @@ public void init(Properties props) {
     if (props.containsKey("line.separator")) {
       lineSeparator = props.getProperty("line.separator").getBytes();
     }
+    if (props.containsKey("key.deserializer")) {
+      try {
+        keyDeserializer =
+            (Deserializer)Class.forName((String) props.get("key.deserializer")).newInstance();
+      } catch (Exception e) {
+        throw new ConfigException("Error initializing Key deserializer", e);
+      }
+    }
   }
 
   @Override
   public void writeTo(ConsumerRecord<byte[], byte[]> consumerRecord, PrintStream output) {
     if (printKey) {
       try {
-        writeTo(consumerRecord.key(), output);
+        if (keyDeserializer != null) {
+          byte[] key = consumerRecord.key() == null ? NULL_BYTES : consumerRecord.key();
+          Object deserializedKey = keyDeserializer.deserialize(null, key);
```

Comment:
you're allowing an arbitrary key deserializer to be specified but in the case of a null key, you're giving this a hard coded specific set of bytes, which it may not know what to do with.